src/app/categories/state/entry-categories.service.ts

```diff
@@ -42,14 +42,13 @@ export class EntryCategoriesService {
   /**
    * Updates the list of categories for the specified entry.
    */
-  updateEntryCategories(entryId: number) {
+  updateEntryCategories(entryId: number, published: boolean) {
```

Comment:
This is works, and can stay if you want, but I would be more inclined to do either:

1. Allow entryId to be null , e.g., `entryId: number | null`
2. Just pass the entry, e.g., `entry: Workflow | Tool` 

Subjective, but it's one less parameter to pass around.

Reply:
I'm going to keep this as it is, I tried implementing both suggestions and ran into some issues...

Allowing the entryId to be `null` is ok until we have to start adding control flow branches to all the call sites, which seems like a bit of a code smell. It also implicitly expects the developer to add this control flow before the call to the service method, which brings potential bugs in the future if someone does not do that.

I liked the idea of passing the entry around, but the primary issue is that `addEntryToCollection` doesn't have a reference to the type of entry, rather it interfaces with the entryID. I would have liked to use this approach because it abstracts the attribute usage of `is_published` down into the service method, but I don't think its doable at the moment.